### PR TITLE
Carousel: disable UI controls for single image

### LINKED ISSUE #20267: Carousel: Remove dot on single image
When someone adds a single image to a gallery and has the carousel enabled, the indicator still shows on the bottom, which might look like a miss-placed character. Maybe we could remove the carousel count for single images? 

annotation-on-2021-07-05-at-12-24-16

Surfaced during testing: p8oabR-GM-p2#comment-5408

### PR DESCRIPTION
**Pull request by ramonjd**
Fixes https://github.com/Automattic/jetpack/issues/20267

#### Changes proposed in this Pull Request

This PR:

- hides navigation arrows and pagination UI if there's only one image.
- disables swipe interaction if there's only one image.
- sets swiper to `enabled` before close. We have to do this because, after testing, it seems as if Swiper doesn't re-enable the navigation arrows on reinitialization.

**Before**

single-image

**After**
Screen Shot 2021-08-02 at 12 10 10 pm

#### Does this pull request change what data or activity we track or use?
Nope.

#### Testing instructions:

Create 2 x galleries on the same post: one with several images and one with only 1 image.

Publish them and checkout the frontend.

You should see and be able to use the navigation arrows and pagination UI for the gallery with several images.

The gallery with one image shouldn't display any UI controls. You shouldn't be able to slide it or interact with the image.

Open one, then the other to make sure we're enabling/disabling the UI correctly.

### CODE REVIEW

File: projects/plugins/jetpack/modules/carousel/jetpack-carousel.js [hunk 1]
```diff
@@ -472,6 +479,10 @@
 					// Fixes some themes where closing carousel brings view back to top.
 					document.documentElement.style.removeProperty( 'height' );
 
+					// Should the navigation arrows if we hid them for single images.
```
Comment (andrewserong) [inline projects/plugins/jetpack/modules/carousel/jetpack-carousel.js:482]:
```suggestion
					// Show the navigation arrows if we hid them for single images.
```
Tiny typo 🙂

Comment (ramonjd) [reply projects/plugins/jetpack/modules/carousel/jetpack-carousel.js]: By blod is typo


😄

**Review by andrewserong — DISMISSED** [review]
Nice change @ramonjd, hides the next/prev buttons and the lone dot in the bottom left 👍 

I noticed there's a very slight flicker where you can see the next/prev buttons for a moment when the carousel is opened, and then again when it's closed:

carousel-single-image-arrows-sml

It's subtle, though, and I suspect most people won't notice, so not a blocker for getting this in.

Tested in Chrome & Safari on iOS and desktop, and FF and Edge on desktop — no lone pagination circle, and no next/prev buttons on a single image in an image gallery. Pagination and next/prev buttons correctly display on a bigger gallery on the same page, and opening/closing between the two works as expected 👍

LGTM!

**Comment by ramonjd** [comment]
@andrewserong Thanks a lot for the review! I think I hadn't pushed the changes I'd intended. 🤕 

I wanted to control the nav + interaction using the Slide API. I've updated the PR and rebased. Hopefully it'll take care of the flashing nav buttons too.

If you have time for another quick review, I'd appreciate it!!

**Comment by andrewserong** [comment]
> If you have time for another quick review, I'd appreciate it!!

Sure thing! Thanks Ramon, I'll give this a whirl later on this morning.

**Review by andrewserong — APPROVED** [review]
This is working nicely for me @ramonjd, re-tested on iOS and desktop browsers on Mac. There's no longer a flash of the next/prev buttons when the carousel is opened. There's still a flash for me when the carousel is closed (presumably because we have to call `swiper.enable()` in `beforeClose` so that it's called before `swiper.destroy()` in `afterClose`. This seems totally reasonable to me, it's such a minor thing, and probably not worth adding in extra CSS / `domUtil.hide` / `show`s just for that case.

LGTM! 🎉

**Review by jeherve — APPROVED** [review]

**Comment by andrewserong** [comment]
@ramonjd looks like this one needs a `git merge origin/master` before it can be merged 🙂

**Comment by ramonjd** [comment]
> @ramonjd looks like this one needs a git merge origin/master before it can be merged 🙂

Thanks for heads up. I'll try to get it in soon before more changes arrive! 😄

**Comment by ramonjd** [comment]
r229791-wpcom

**Pull request merged**